js/initcontext.go

```diff
@@ -70,7 +70,7 @@ func NewInitContext(
 		moduleRegistry:    getJSModules(),
 		exportsCache:      make(map[string]goja.Value),
 		moduleVUImpl: &moduleVUImpl{
-			ctx:     context.Background(),
+			ctx:     context.Background(), // TODO: pass a real context here as well?
```

Comment:
```suggestion
			ctx:     context.Background(), // TODO: pass a real context as we did for https://github.com/grafana/k6/pull/2800
```
It think it should be even better if we open a dedicated issue. Or does one of the next PR  resolve it?

Reply:
None of my chain of PRs resolve this, I didn't have time and it's not super straightforward to do so, there are some chicken-and-egg problems and some tricky things to consider when it comes to test suites (https://github.com/grafana/k6/issues/1342). 

Still, it is necessary to solve that before https://github.com/grafana/k6/issues/2804 is implemented, so I'd rather add a link to both that issue and the #2800 PR as well :+1:

Reply:
```suggestion
			// TODO: pass a real context as we did for https://github.com/grafana/k6/pull/2800,
			// also see https://github.com/grafana/k6/pull/2800
			ctx:     context.Background(), 
```